File: Maths/factorielle.js
/**
 * La factorielle d'un entier n est le produit des nombres de 1 à n.
 * Méthode récursive.
 * @param {number} int
 * @returns {number} factorielle de int
 */
const rFactorielle = (int) => (int <= 1 ? 1 : int * rFactorielle(int - 1));

/**
 * La factorielle d'un entier n est le produit des nombres de 1 à n.
 * Méthode itérative.
 * @param {number} int
 * @returns {number} factorielle de int
 */
const iFactorielle = (int) => {
  let result = 1;
  for (let i = 2; i <= int; i++) {
    result *= i;
  }
  return result;
};

// Test de rapidité pour les deux méthodes

// méthode récursive pour n = 100:  3.5ms
console.time("rFactorielle");
console.log(rFactorielle(100));
console.timeEnd("rFactorielle");

// méthode itérative pour n = 100:  0.08ms
console.time("iFactorielle");
console.log(iFactorielle(100));
console.timeEnd("iFactorielle");

File: Python/Maths/factorielle.py
def factorielle_rec(n: int) -> int:
    """
    Description:
        Factorielle méthode récursive
    
    Paramètres:
        n: {int} -- Nombre à factorielle
    
    Retourne:
        {int} -- Factorielle de n
    
    Exemple:
        >>> factorielle_rec(100)
        9.332622e+157

        Pour l'écriture scientifique: f"{factorielle_rec(100):e}"
    """
    return 1 if n == 0 else n * factorielle_rec(n - 1)

def factorielle_it(n: int) -> int:
    """
    Description:
        Factorielle méthode itérative
    
    Paramètres:
        n: {int} -- Nombre à factorielle
    
    Retourne:
        {int} -- Factorielle de n
    
    Exemple:
        >>> factorielle_it(100)
        9.332622e+157

        Pour l'écriture scientifique: f"{factorielle_it(100):e}"
    """
    result = 1
    for i in range(1, n + 1):
        result *= i
    return result